### PR TITLE
Add RowData Serializer

### PR DESCRIPTION
**Pull request by InigoSJ**
Add RowData Serializer and examples. 

As a reminder, we cannot get the schema from a RowData element (as we do with GenericRecord), so we need the schema when initializing the Serializer and store the data types for future serialization

b/383889302

### CODE REVIEW

File: connectors/bigtable/flink-connector-gcp-bigtable/src/main/java/com/google/flink/connector/gcp/bigtable/utils/ErrorMessages.java [hunk 1]
```diff
@@ -29,8 +29,14 @@ public class ErrorMessages {
...
+    public static final String NESTED_TYPE_ERROR =
+            "Nested rows are only supported with withNestedRowsMode and not double nested. Use Bytes for more complex types";
+    public static final String ROW_KEY_STRING_TYPE = "Row Key has to be of type String, got ";
     public static final String SERIALIZER_ERROR =
             "Error while serializing element to RowMutationEntry: ";
     public static final String METRICS_ENTRY_SERIALIZATION_WARNING =
             "Error while serializing RowMutationEntry for metrics, entry will be counted as 0 bytes. This error doesn't affect the job. Error: ";
+    public static final String TIMESTAMP_OUTSIDE_PRECISION_TEMPLATE =
+            "Unsupported precision, needs to be between %s and %s, got %s";
+    public static final String MISSING_ROW_KEY = "Row key field %s not found in schema %s";
```
Comment (clmccart) [inline connectors/bigtable/flink-connector-gcp-bigtable/src/main/java/com/google/flink/connector/gcp/bigtable/utils/ErrorMessages.java:41]: this is also a template

File: connectors/bigtable/flink-connector-gcp-bigtable/src/main/java/com/google/flink/connector/gcp/bigtable/serializers/RowDataToRowMutationSerializer.java [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+                        BigtableUtils.getTimestamp(context),
+                        ByteString.copyFrom(valueBytes));
+            }
+        }
+        return entry;
+    }
+
+    /**
+     * Generates maps from the schema to keep track of DataTypes and indexes. This is needed since
+     * the schema cannot be fetch from the {@link RowData}.
```
Comment (clmccart) [inline connectors/bigtable/flink-connector-gcp-bigtable/src/main/java/com/google/flink/connector/gcp/bigtable/serializers/RowDataToRowMutationSerializer.java:161]: nit: fetched

File: connectors/bigtable/flink-connector-gcp-bigtable/src/test/java/com/google/flink/connector/gcp/bigtable/serializers/RowDataTest.java [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+                    DataTypes.FIELD(
+                            TestingUtils.NESTED_COLUMN_FAMILY_2,
+                            DataTypes.ROW(
+                                    DataTypes.FIELD(
+                                            TestingUtils.STRING_FIELD_2, DataTypes.STRING()))));
+
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    public void testCorrectSerializerInitialization(Boolean useNestedRowsMode) {
+        RowDataToRowMutationSerializer.Builder builder =
```
Comment (clmccart) [inline connectors/bigtable/flink-connector-gcp-bigtable/src/test/java/com/google/flink/connector/gcp/bigtable/serializers/RowDataTest.java:80]: can we also add a test for null values on all the initialization configuration fields?

File: connectors/bigtable/flink-connector-gcp-bigtable/src/test/java/com/google/flink/connector/gcp/bigtable/serializers/RowDataTest.java [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+                        String.format(
+                                ErrorMessages.TIMESTAMP_OUTSIDE_PRECISION_TEMPLATE,
+                                DataTypes.TIMESTAMP(
+                                        RowDataToRowMutationSerializer.MAX_DATETIME_PRECISION + 1),
+                                RowDataToRowMutationSerializer.MIN_DATETIME_PRECISION,
+                                RowDataToRowMutationSerializer.MAX_DATETIME_PRECISION));
+    }
+
+    @Test
+    public void testRowKeyNoStringerror() {
```
Comment (clmccart) [inline connectors/bigtable/flink-connector-gcp-bigtable/src/test/java/com/google/flink/connector/gcp/bigtable/serializers/RowDataTest.java:186]: nit: capitalize the E in error

File: connectors/bigtable/flink-connector-gcp-bigtable/src/test/java/com/google/flink/connector/gcp/bigtable/serializers/RowDataTest.java [hunk 5]
```diff
@@ -0,0 +1,469 @@
...
+                                        .withSchema(dtIntegerKeySchema)
+                                        .withRowKeyField(TestingUtils.ROW_KEY_FIELD)
+                                        .withColumnFamily(TestingUtils.COLUMN_FAMILY)
+                                        .build())
+                .hasMessage(
+                        String.format(ErrorMessages.ROW_KEY_STRING_TYPE_TEMPLATE, DataTypes.INT()));
+    }
+
+    @Test
+    public void testMissingRowKey() {
```
Comment (clmccart) [inline connectors/bigtable/flink-connector-gcp-bigtable/src/test/java/com/google/flink/connector/gcp/bigtable/serializers/RowDataTest.java:202]: out fo curiosity, what's the behavior if the rowkey is a field in a nested row?

**Review by clmccart — DISMISSED** [review]

File: connectors/bigtable/flink-connector-gcp-bigtable/src/test/java/com/google/flink/connector/gcp/bigtable/serializers/RowDataTest.java (hunk 5 above)
Comment (InigoSJ) [reply connectors/bigtable/flink-connector-gcp-bigtable/src/test/java/com/google/flink/connector/gcp/bigtable/serializers/RowDataTest.java:202]: It will fail for not being a string

**Comment by clmccart** [comment]
closable bug to description :)

File: connectors/bigtable/flink-connector-gcp-bigtable/src/test/java/com/google/flink/connector/gcp/bigtable/serializers/RowDataTest.java (hunk 3 above)
Comment (InigoSJ) [reply connectors/bigtable/flink-connector-gcp-bigtable/src/test/java/com/google/flink/connector/gcp/bigtable/serializers/RowDataTest.java:80]: Added two test for null key and null schema.  I did the same for GenericRecord
